### PR TITLE
`bun test` auto set `NODE_ENV=test`

### PR DESCRIPTION
**Pull request by Parzival-3141**
### What does this PR do?
Closes https://github.com/oven-sh/bun/issues/4118

Automatically sets `NODE_ENV=test` when starting up the test runner. Currently doesn't overwrite a preexisting value.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
I wrote automated tests

- [x] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files
- [x] I included a test for the new code, or an existing test covers it

### CODE REVIEW

**Comment by Parzival-3141** [comment]
Currently causing an issue in `webpack.test.ts` where the `NODE_ENV=test` gets propagated to the new `bun` process.
https://github.com/oven-sh/bun/blob/f267c1d097923a2d2992f9f60a6dd365fe706512/test/js/third_party/webpack/webpack.test.ts#L9-L16

File: test/cli/test/bun-test.test.ts [hunk 1]
```diff
@@ -693,6 +693,21 @@ describe("bun test", () => {
...
   });
+  // https://github.com/oven-sh/bun/issues/4118
+  test("expect NODE_ENV == 'test'", () => {
+    const stderr = runTest({
+      input: `
+        import {expect, test} from "bun:test";
+        test("expect NODE_ENV=test", () => {
+            expect(process.env.NODE_ENV).toBe("test");
+            expect(process?.env.NODE_ENV).toBe("test");
+            expect(process && process.env && process.env.NODE_ENV).toBe("test");
```
Comment (Electroid) [inline test/cli/test/bun-test.test.ts]: Don't these all test the same thing? The first line is enough.

Comment (Parzival-3141) [reply test/cli/test/bun-test.test.ts]: This was the repro on the issue. They were mainly concerned with the optional chaining example.

**Comment by paperclover** [comment]
closing as the same bug was included in some other pr, though we should hvae merged this when it was opened

**Pull request closed without merging**